src/modelinstance.cpp

```diff
@@ -181,77 +142,145 @@ Status validateConfigurationAgainstNetwork_2(const ModelConfig& config, std::sha
-        }
-    } else if (config.getLayouts().size() > 0) {
-        ov::preprocess::PrePostProcessor preproc(network);
-        for (const auto& [name, layout] : config.getLayouts()) {
-            if (hasInputWithName(network, name)) {
-                std::string networkLayout = layout == "NCHW" ? "NHWC" : "NCHW";
-                SPDLOG_LOGGER_DEBUG(modelmanager_logger, "model: {}, version: {}; Adding preprocessing step: {}; Tensor Layout:{}; Network Layout:{}; input name: {}", modelName, modelVersion, layout, layout, networkLayout, name);
-                preproc.input(name).tensor().set_layout(ov::Layout(layout));
-                preproc.input(name).model().set_layout(ov::Layout(networkLayout));
-            }
-            if (hasOutputWithName(network, name)) {
-                std::string networkLayout = layout == "NCHW" ? "NHWC" : "NCHW";
-                SPDLOG_LOGGER_DEBUG(modelmanager_logger, "model: {}, version: {}; Adding preprocessing step: {}; Tensor Layout:{}; Network Layout:{}; output name: {}", modelName, modelVersion, layout, layout, networkLayout, name);
-                preproc.output(name).tensor().set_layout(ov::Layout(layout));
-                preproc.output(name).model().set_layout(ov::Layout(networkLayout));
+            std::string name = input.get_any_name();
+
+            if (config.getLayout_2().isSet()) {
+                SPDLOG_LOGGER_DEBUG(modelmanager_logger, "model: {}, version: {}; Adding preprocessing step: Tensor Layout:{}; Network Layout:{}; single input",
+                    modelName,
+                    modelVersion,
+                    config.getLayout_2().getTensorLayout(),
+                    config.getLayout_2().getModelLayout());
+
+                // TODO: Validate rank vs layout string len?
+                preproc.input().tensor().set_layout(ov::Layout(config.getLayout_2().getTensorLayout()));
+                preproc.input().model().set_layout(ov::Layout(config.getLayout_2().getModelLayout()));
+            } else if (config.getLayouts_2().count(name) > 0) {
+                auto& layout = config.getLayouts_2().at(name);
+                SPDLOG_LOGGER_DEBUG(modelmanager_logger, "model: {}, version: {}; Adding preprocessing step: Tensor Layout:{}; Network Layout:{}; input name: {}",
+                    modelName,
+                    modelVersion,
+                    layout.getTensorLayout(),
+                    layout.getModelLayout(),
+                    name);
+
+                // TODO: Validate rank vs layout string len?
+                preproc.input(name).tensor().set_layout(ov::Layout(layout.getTensorLayout()));
+                preproc.input(name).model().set_layout(ov::Layout(layout.getModelLayout()));
+            } else {
+                std::string guessedModelLayout{"N..."};
+                SPDLOG_LOGGER_DEBUG(modelmanager_logger, "model: {}, version: {}; Adding auto preprocessing step: Tensor Layout:; Network Layout:{}; input name: {}",
```

Comment:
missing {} after Tensor Layout:

Reply:
When --layout is not specified, the default value is empty Tensor Layout and "N..." Model Layout. This is why {} is missing, because tensor layout is empty.